File: backend/src/models/comment.js
const mongoose = require('mongoose');

const CommentSchema = new mongoose.Schema({
  user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
  },
  post: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Post',
  },
  text: {
    type: String,
    required: true,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model('Comment', CommentSchema);

File: frontend/lib/models/comment.dart

import 'package:jivvi/models/user.dart';

class Comment {
  final String id;
  final String postId;
  final User author;
  final String text;
  final DateTime timestamp;

  Comment({
    required this.id,
    required this.postId,
    required this.author,
    required this.text,
    required this.timestamp,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json['_id'] ?? json['id'] ?? '',
      postId: json['post'] ?? '', // Assuming the API returns 'post' field for postId
      author: User.fromJson(json['author'] is String 
          ? {'_id': json['author']} // Handle case where author might just be an ID
          : (json['author'] as Map<String, dynamic>)),
      text: json['text'] ?? '',
      timestamp: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : DateTime.now(),
    );
  }
}
